### DISCUSSION TITLE
ModbusClientRTU: Simultaneous use of sync request and async request

### DISCUSSION THREAD

**Discussion by MPeter74** [eModbus/eModbus#374] [Q&A]
Is it possible to use the api for async requests and sync requests “simultaneously” for the ModbusClientRTU in an application?
E.g. in multitasking systems: One task uses the blocking sync request API. Another task uses the non-blocking async requests.
Both parts of the application would use the same instance of the ModbusClientRTU.

**Comment by bertmelis** [comment]
I never tested this. I don't see any direct reason why it wouldn't work besides memory. Maybe some underlying stack might cause issues?

Did you try it?

**Comment by MPeter74** [comment]
Yes I tried it, and it seems to work.

But due to time constraints, I have not “seriously” tested it for any edge cases that are not required in our application.

I'm just not sure if it's an undocumented feature or if it runs randomly and could cause problems at some point in future?

**Comment by bertmelis** [comment]
Wait, I'm on a different path. You are using the blocking and non-blocking API from the same client.

Yes, it should work. The blocking request will wait for its turn in the queue.

**Comment by MPeter74** [comment]
Yes, I'm using the blocking (sync requests) and non blocking (async requests) on the same instance of  ModbusClientRTU.

So you think this should work without any problems. (Therefore it is an undocumented feature ;-))
